UserInterface/GameManager/GameManager.cs

```diff
@@ -67,45 +88,88 @@ private void RunGame()
             while (isGameOnGoing)
             {
                 Console.CursorVisible = false;
-                Pair? currentPair = _pairManager.Pair;
 
-                _board.FillBoardWithAnimals(_animals);
+                _board.FillBoardWithObjects(_gameObjects);
 
                 _userInterface.DrawBoard(_board);
-                _board.RemoveAnimalFromBoard(_animals);
-
-                _animalActions.Move(_board, _animals);
+                _board.PrepareBoard(_gameObjects);
+                
+                _movementManager.Act(_gameObjects, _board);
 
-                _pairManager.RemoveNonExistsPairs(_pairs);
-
-                _pairManager.CheckPairForExistence(_pairs, _board, _animals, _children);
+                if (_pairs.Count != 0)
+                {
+                    _pairManager.RemoveNotExistingPairs(_pairs);
+                }
+               
+                _pairManager.CheckPairForExistence(_pairs, _board, _gameObjects, _childrenGameObjects);
 
-                _pairManager.CreatePair(_animals);
+                Pair newPair = _pairManager.CreatePair(_gameObjects);
 
-                if (currentPair != null)
+                if (newPair != null)
```

Comment:
Here similar to comment above you are implementing some check in this method although that should be implemented in class which manages that object.

So checking and adding pairs should be inside _pair class. Either in CreatePaier() method or in case you need to use that method sometimes without adding pairs then you should create new method in pair class that does both tasks and then that method should be used in this method.